File: Implementierung/Backend/src/main/java/edu/kit/informatik/model/userdata/interactions/Interaction.java
package edu.kit.informatik.model.userdata.interactions;

import edu.kit.informatik.model.DataObject;
import edu.kit.informatik.model.User;
import edu.kit.informatik.model.userdata.courses.Session;
import jakarta.persistence.Entity;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.sql.Timestamp;

/**
 * Entität einer Interaktion
 *
 * @author uqkai
 * @author ugqbo
 * @version 1.0
 */
@Getter
@NoArgsConstructor
@Entity
@Table(name = "interactions")
public class Interaction extends DataObject {

    private String timeStamp;

    @ManyToOne
    private Session session;

    @ManyToOne
    private Participant from;

    @ManyToOne
    private Participant to;

    @ManyToOne
    private Category category;

    /**
     * @param id Id
     * @param user {@link User}
     * @param timeStamp Zeitstempel
     * @param session {@link Session Sitzung}
     * @param from {@link Participant From-Participant} von dem die Interaktion ausgeht
     * @param to {@link Participant To-Participant} zu dem die Interaktion geht
     * @param category {@link Category Kategorie}
     * @param createdAt {@link Timestamp} der Erstellung
     * @param updatedAt {@link Timestamp} der letzten Änderung
     */
    public Interaction(String id, User user, String timeStamp, Session session, Participant from, Participant to,
                       Category category, Timestamp createdAt, Timestamp updatedAt) {
        super(id, user, createdAt, updatedAt);
        this.timeStamp = timeStamp;
        this.session = session;
        this.from = from;
        this.to = to;
        this.category = category;
    }
}

File: Implementierung/frontend/paschi/src/model/userdata/interactions/Interaction.ts
import {User} from "@/model/User";
import {Participant} from "@/model/userdata/interactions/Participant";
import {Category} from "@/model/userdata/interactions/Category";
import {DataObject} from "@/model/DataObject";
import {Session} from "@/model/userdata/courses/Session";

export class Interaction extends DataObject {

  private readonly _user: User;
  private readonly _timeStamp: string;
  private readonly _session: Session;
  private _fromParticipant: Participant;
  private _toParticipant: Participant;
  private _category: Category;

  constructor(id: string | undefined, localId: number, user: User, timeStamp: string, session: Session,
              fromParticipant: Participant, toParticipant: Participant, category: Category) {
    super(id, localId);
    this._user = user;
    this._timeStamp = timeStamp;
    this._session = session;
    this._fromParticipant = fromParticipant;
    this._toParticipant = toParticipant;
    this._category = category;
  }

  get user(): User {
    return this._user;
  }

  get timeStamp(): string {
    return this._timeStamp;
  }

  get session(): Session {
    return this._session;
  }

  get fromParticipant(): Participant {
    return this._fromParticipant;
  }

  set fromParticipant(value: Participant) {
    this._fromParticipant = value;
    this.update();
  }

  get toParticipant(): Participant {
    return this._toParticipant;
  }

  set toParticipant(value: Participant) {
    this._toParticipant = value;
    this.update();
  }

  get category(): Category {
    return this._category;
  }

  set category(value: Category) {
    this._category = value;
    this.update();
  }
}
